ledger/participant-integration-api/src/main/scala/platform/store/backend/common/Field.scala

```diff
@@ -91,11 +92,21 @@ private[backend] case class BooleanOptional[FROM](extract: FROM => Option[Boolea
   override def convert: Option[Boolean] => lang.Boolean = _.map(Boolean.box).orNull
 }
 
+private[backend] object Timestamp {
+  def instantToMicros(i: Instant): Long =
+    TimeUnit.SECONDS.toMicros(i.getEpochSecond) + TimeUnit.NANOSECONDS.toMicros(i.getNano.toLong)
+}
+
 private[backend] case class Timestamp[FROM](extract: FROM => Instant)
-    extends TrivialField[FROM, Instant]
+    extends Field[FROM, Instant, java.lang.Long] {
```

Comment:
the java type is only needed to force non primitive transport to signall nulls. this could be simply Long I think